### PR TITLE
Add company fields to campaign condition

### PR DESCRIPTION
**Pull request by kuzmany**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/6674
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

Bugfix/feature request donate by https://www.testin.cn/ 

[//]: # ( Required: )
#### Description:

Company fields added to campaign contact field condition

image

#### Steps to test this PR:
1.  Create campaign with condition and action
2.  Create condition based on contact company fields (for example Company Address 1). 
Also  create condition based on contact field (Address 1 or email or Firstname or whatever)
3. Run campaign. Test if condition based on contact or company fields works properly

### CODE REVIEW

**Comment by npracht** [comment]
Label: Pending test confirmation

**Comment by javjim-mautic** [comment]
Working properly thanks!

**Comment by YosuCadilla** [comment]
Created a new Contact (myself)
Created a new Segment where email = me
Created a new campaign with condition = (right when I selected core company the page refreshed and thrown this:)
"500 internal server error - php notice - undefined offset:1"

**Comment by kuzmany** [comment]
@YosuCadilla any detail logs? Did you test it on dev enviroment?

**Comment by kuzmany** [comment]
@YosuCadilla  I see it on slack. Thanks

**Comment by kuzmany** [comment]
@YosuCadilla that error see if you save campaign condition ? Or on the run mautic:trigger:campaign? Or when? Need better  steps to reproduce.

**Comment by dongilbert** [comment]
@kuzmany I think he was testing in Mautibox @ https://mautibox.com/5628/

**Comment by YosuCadilla** [comment]
@kuzmany  When creating a new campaign, right when adding the condition.
Yes, I am using mautibox, the screen that loads automatically is in response of an error in the background. I guess it would just fail silently (add some to the log)  on a regular install?

**Comment by kuzmany** [comment]
I created test campaign on Mautibox and works for me. 
Please can you create your campaing and re-test? I need see it

**Comment by kuzmany** [comment]
Got it. Working on it

**Comment by kuzmany** [comment]
@YosuCadilla fixed it by last commit. I don't know when Mauticbox will rebuild with actual content. What @heathdutton ?

**Comment by YosuCadilla** [comment]
Thank you @kuzmany.
Please @heathdutton mention me when I can proceed with further testing, thank you.

**Comment by heathdutton** [comment]
Mautibox will prompt the rebuild ~5min after a PR is updated. You'll see a notice when it's rebuilding and then it will refresh the screen when done. You should be able to test here @YosuCadilla https://mautibox.com/5628

**Comment by kuzmany** [comment]
PR reworked.
Guys, can you re-test?

**Comment by Woeler** [comment]
@alanhartless @heathdutton How's your reviews on this now?

File: app/bundles/LeadBundle/Entity/LeadFieldRepository.php [hunk 1]
```diff
@@ -116,11 +116,43 @@ public function getFieldAliases($object = 'lead')
...
                 ->orderBy('f.field_order', 'ASC')
                 ->execute()->fetchAll();
     }
 
+    /**
+     * Add company left join.
+     *
+     * @param \Doctrine\ORM\QueryBuilder|\Doctrine\DBAL\Query\QueryBuilder $q
+     */
+    public function addCompanyLeftJoin($q)
```
Comment (alanhartless) [inline app/bundles/LeadBundle/Entity/LeadFieldRepository.php]: I don't see this or getPropertyByField used outside this class. Should they be private?

Comment (kuzmany) [reply app/bundles/LeadBundle/Entity/LeadFieldRepository.php]: Change to private.

**Comment by Woeler** [comment]
@kuzmany I have pinged @heathdutton and @alanhartless to re-check here. Then we can get this in.

**Pull request merged**